pkg/clustermanager/cluster_manager_test.go

```diff
@@ -905,12 +905,48 @@ func TestClusterManagerResumeEKSAControllerReconcileSuccessWithoutMachineConfig(
 }
 
 func TestClusterManagerInstallCustomComponentsSuccess(t *testing.T) {
+	features.ClearCache()
 	oldCloudstackProviderFeatureValue := os.Getenv(features.CloudStackProviderEnvVar)
 	err := os.Unsetenv(features.CloudStackProviderEnvVar)
-	defer os.Setenv(features.CloudStackProviderEnvVar, oldCloudstackProviderFeatureValue)
 	if err != nil {
-		return
+		t.Errorf("error unsetting env var %v", features.CloudStackProviderEnvVar)
+	}
+	defer func() {
+		err := os.Setenv(features.CloudStackProviderEnvVar, oldCloudstackProviderFeatureValue)
+		if err != nil {
+			t.Errorf("error setting env var %v", features.CloudStackProviderEnvVar)
+		}
+	}()
+	ctx := context.Background()
+	tt := newTest(t)
+	tt.clusterSpec.VersionsBundle.Eksa.Components.URI = "testdata/testClusterSpec.yaml"
+
+	tt.mocks.client.EXPECT().ApplyKubeSpecFromBytes(tt.ctx, tt.cluster, gomock.Not(gomock.Nil())).Return(nil)
+
+	for namespace, deployments := range internal.EksaDeployments {
+		for _, deployment := range deployments {
+			tt.mocks.client.EXPECT().WaitForDeployment(ctx, tt.cluster, "30m", "Available", deployment, namespace)
+		}
+	}
+	tt.mocks.provider.EXPECT().InstallCustomProviderComponents(ctx, tt.cluster.KubeconfigFile)
+	if err := tt.clusterManager.InstallCustomComponents(tt.ctx, tt.clusterSpec, tt.cluster, tt.mocks.provider); err != nil {
+		t.Errorf("ClusterManager.InstallCustomComponents() error = %v, wantErr nil", err)
+	}
+}
+
+func TestClusterManagerInstallCustomComponentsSuccessWithFullLifecycleAPI(t *testing.T) {
+	features.ClearCache()
+	oldFullLifecycleAPIFeatureValue := os.Getenv(features.FullLifecycleAPIEnvVar)
+	err := os.Setenv(features.FullLifecycleAPIEnvVar, "true")
+	if err != nil {
+		t.Errorf("error setting env var %v to true", features.FullLifecycleAPIEnvVar)
 	}
+	defer func() {
+		err := os.Setenv(features.FullLifecycleAPIEnvVar, oldFullLifecycleAPIFeatureValue)
```

Comment:
something I learn from dan, you can now do `t.SetEnv` to set an env var only for a test duration

Reply:
TIL